rbd/admin/admin.go

```diff
@@ -0,0 +1,63 @@
+// +build !nautilus
+
+package admin
+
+import (
+	"fmt"
+
+	ccom "github.com/ceph/go-ceph/common/commands"
+)
+
+// RBDAdmin is used to administrate rbd volumes and pools.
+type RBDAdmin struct {
+	conn ccom.RadosCommander
+}
+
+// NewFromConn creates an new management object from a preexisting
+// rados connection. The existing connection can be rados.Conn or any
+// type implementing the RadosCommander interface.
+func NewFromConn(conn ccom.RadosCommander) *RBDAdmin {
+	return &RBDAdmin{conn}
+}
+
+// LevelSpecifier provides an interface that can be used where ceph requires a
+// levelspec.
+type LevelSpecifier interface {
+	levelSpec() string
+}
+
+// RawLevelSpec can be used to pass a level spec directly to ceph without
+// constructing it from component values. This should only be used if LevelSpec
+// can not represent the levelspec value you want to pass to ceph.
+type RawLevelSpec string
+
+func (r RawLevelSpec) levelSpec() string {
+	return string(r)
+}
+
+// LevelSpec is used to construct a levelspec given a pool and
+// optional Namespace and Image names.
+type LevelSpec struct {
+	Pool      string
+	Namespace string
+	Image     string
+}
+
+func (ls LevelSpec) levelSpec() string {
+	return ls.String()
+}
+
+// String returns a string representation of the LevelSpec.
+func (ls LevelSpec) String() string {
+	var s string
+	if ls.Image != "" && ls.Namespace != "" {
+		s = fmt.Sprintf("%s/%s/%s", ls.Pool, ls.Namespace, ls.Image)
+	} else if ls.Image != "" {
+		s = fmt.Sprintf("%s/%s", ls.Pool, ls.Image)
+	} else if ls.Namespace != "" {
+		s = fmt.Sprintf("%s/%s/", ls.Pool, ls.Namespace)
+	} else {
+		s = fmt.Sprintf("%s/", ls.Pool)
```

Comment:
`/` slash after the pool is not required here? and also above when a namespace is provided?

Reply:
I based the Go implementation on this function in ceph:
https://github.com/ceph/ceph/blob/a67d1cf2a7a4031609a5d37baa01ffdfef80e993/src/tools/rbd/Schedule.cc#L69

The python code in src/pybind/mgr/rbd_support/schedule.py, does something similar too (although its not as obvious IMO).

If you have reason to think the trailing slashes are *wrong* I'd like to know sooner rather than later. I find it a bit annoying that I didn't find any docs on the format despite it being commonly used in some of the apis. Thus I had to resort to reading code.

Reply:
I was just going through the CLI commands we use, looks like it's fine to use '/' as per C++ code,  Thanks @phlogistonjohn

Reply:
OK, do you use `ceph` command directly or `rbd`? Because `rbd` constructs the levelspec based on other cli options whereas `ceph` expects a levelspec already "formatted".